src/java.base/share/classes/sun/security/ssl/SSLTransport.java

```diff
@@ -122,16 +123,42 @@ static Plaintext decode(TransportContext context,
         } catch (AEADBadTagException bte) {
             throw context.fatal(Alert.BAD_RECORD_MAC, bte);
         } catch (BadPaddingException bpe) {
-            /*
-             * The basic SSLv3 record protection involves (optional)
-             * encryption for privacy, and an integrity check ensuring
-             * data origin authentication.  We do them both here, and
-             * throw a fatal alert if the integrity check fails.
-             */
-             Alert alert = (context.handshakeContext != null) ?
-                     Alert.HANDSHAKE_FAILURE :
-                     Alert.BAD_RECORD_MAC;
-            throw context.fatal(alert, bpe);
+            // Check for unexpected plaintext alert message during TLSv1.3 handshake.
+            // This can happen if client doesn't receive ServerHello due to network timeout
+            // and tries to close the connection by sending an alert message.
+            if (srcsLength == 1 && !context.sslConfig.isClientMode && !context.isNegotiated &&
+                    context.handshakeContext != null &&
+                    context.handshakeContext.negotiatedProtocol.useTLS13PlusSpec()) {
+                ByteBuffer packet = srcs[srcsOffset].duplicate();
+                packet.position(0);
+                byte contentType = (byte) Record.getInt8(packet);                   // pos: 0
+                byte majorVersion = (byte) Record.getInt8(packet);                  // pos: 1
+                byte minorVersion = (byte) Record.getInt8(packet);                  // pos: 2
+                int contentLen = Record.getInt16(packet);                           // pos: 3, 4
+
+                if (contentLen == 2 && ContentType.ALERT.equals(ContentType.valueOf(contentType))) {
+                    if (SSLLogger.isOn && SSLLogger.isOn("ssl")) {
+                        SSLLogger.finest("Processing plaintext alert during TLSv1.3 handshake");
+                    }
+
+                    plaintexts = new Plaintext[]{
```

Comment:
do we need to advance the position of the input buffer here?

Reply:
You mean the `packet` buffer? No, it has 2 bytes remaining as it should.

Reply:
I was referring to `srcs[srcOffset]`; `packet` is a duplicate, so the position is independent from the original.

Reply:
No, the position was already advanced in `decodeInputRecord`. The test correctly reports `bytesConsumed = 7` when server unwraps the alerts.

Reply:
ah, I see that now.